### DISCUSSION TITLE
Resizing an element like in freeTransform plugin

### DISCUSSION THREAD

**Discussion by sok82** [clientIO/joint#1662] [Q&A] [answered]
Hi all
I'm working on my own diagram editor based on Joint JS core.
Implemented a bunch of different elements and now is the time for resizing of elements.

I found that there is no such an out-of-the-box tool in JointJS core, only [FreeTransform plugin](https://resources.jointjs.com/docs/rappid/v2.1/ui.html#ui.FreeTransform) , which is in JointJS+, that is not affordable for my project.

I'm looking for a way to implement resizing by myself, but struggling with how to start.
I need some common way to resize all different shapes and do not know how to implement it in a form of a tool \ plugin.
I need
1. Render some boundary which would be responsive to dragging sides of it with mouse pointer
2. Detect mouse move
3. Calling resize on mouse moves

I found something about resizing here - https://github.com/Rodsevich/JointTooledViewPlugin/blob/master/tooledViewPlugin.js and common idea is clear (detecting mousemove and calling resize on mouseup).

But how to implement some common routine that could be attached to any shape? 

Any ideas would be appreciated

**Comment by kumilingus** [comment, marked as answer]
You can use the [`elementTools.Control`](https://resources.jointjs.com/docs/jointjs/v3.5/joint.html#elementTools.Control) as shown in this example:

https://codepen.io/jointjs/pen/poaEpKx

Or implement your own [`elementTool`](https://resources.jointjs.com/tutorial/element-tools) (by inheriting from `dia.ToolView`) similar to [this](https://github.com/clientIO/joint/blob/master/src/elementTools/control.mjs).

**Reply by sok82** [reply]
Thanks a lot, Roman
Looks like  good point to start from
